### ISSUE TITLE
Phase 2: Connectors, Sync & Core Module Wiring

### ISSUE THREAD

**Issue by ChristianV997** [ChristianV997/my_OS#59]
## Context

Phase 1 (loop integration) is **complete and merged**: error recovery, `strategy_memory`, `auto_kill`/`amplifier`, real Shopify/Meta data, research scheduler daemon, and regime transition handling are all wired in `backend/execution/loop.py`, `backend/decision/engine.py`, and `backend/api.py`.

Three gaps remain before the system is production-ready:

1. **Branch is behind main** — PRs #54, #56, #58 merged LinUCB contextual bandit, TikTok connector, causal model, and portfolio engine to `main` after our branch diverged.
2. **Existing connectors are disconnected** — `connectors/meta_ads_intel.py`, `connectors/shopify_scraper.py`, and the entire `core/` discovery loop (embeddings, intelligence, CAC) have no callers in `backend/`.
3. **No production-grade external connectors** — Supabase (persistent DB), FRED/S&P macro signals, and Adobe Journey Optimizer (real campaign pause/scale) are absent.

---

### Step 1 — Sync with main
```bash
git fetch origin main && git merge origin/main
```
- Resolve conflicts in `backend/execution/loop.py` and `backend/decision/engine.py` (keep Phase 1 wiring, integrate LinUCB if present)
- After merge: replace `bandit_weight()` with LinUCB `predict(action, context)` in `backend/decision/engine.py` if `core/contextual_bandit.py` is present

---

### Step 2 — Wire `core/` discovery loop into backend

**`backend/api.py` — `_research_runner()`**: after `scheduler.tick()`, pass trend keywords to `core/intelligence_loop.run_intelligence()` so Google Trends data feeds the opportunity discovery engine.

**`backend/api.py` — `/metrics` endpoint**: surface `cac_estimate` from `core/cac.estimate_cac(get_memory())`.

**`backend/decision/scoring.py` — `causal_score()`**: add `competition_penalty` using `connectors/meta_ads_intel.MetaAdsIntel().competition_score()` when `META_ACCESS_TOKEN` is set. High-competition variants score lower.

---

### Step 3 — Supabase connector (Priority 1)

**New file: `backend/integrations/supabase_client.py`**
- `save_state(state_dict)` — upserts to `system_state` table
- `load_state()` — returns last saved state dict
- `append_events(rows)` — bulk inserts to `events` table

**Modify `backend/core/serializer.py`**: dual-write to Supabase when `SUPABASE_URL` is set; fall back to DuckDB for local dev.

**Required Supabase tables (run once in dashboard):**
```sql
create table system_state (id text primary key, data jsonb, updated_at timestamptz default now());
create table events (id bigserial primary key, data jsonb, created_at timestamptz default now());
```

**New env vars:** `SUPABASE_URL`, `SUPABASE_SERVICE_ROLE_KEY`

---

### Step 4 — FRED / S&P macro signals (Priority 2)

**New file: `backend/integrations/macro_signals.py`**
- `get_macro_signals()` — fetches GDP growth, inflation (CPI), consumer sentiment (UMCSENT), and VIX from FRED API; cached in-process

**Modify `backend/regime/detector.py`**: apply macro override after base regime detection:
- VIX > 30 → force `"volatile"`
- Negative GDP growth + base `"stable"` → force `"decay"`

**New env vars:** `FRED_API_KEY` (free at fred.stlouisfed.org), `ICE_API_KEY` (optional S&P Global premium)

---

### Step 5 — Adobe Journey Optimizer (Priority 3)

**New file: `backend/integrations/adobe_ajo.py`**
- `pause_campaign(campaign_id)` — PATCH `/journey/campaigns/{id}` with `{"status": "stopped"}`
- `scale_campaign(campaign_id, budget_multiplier)` — PATCH with `{"budgetMultiplier": x}`

**Modify `backend/execution/loop.py`**:
- After `_kill_set.add(campaign_id)` → call `adobe_ajo.pause_campaign()` when `ADOBE_AJO_TOKEN` is set
- After `state.memory.extend(amplified[-5:])` → call `adobe_ajo.scale_campaign()` for each amplified winner

**New env vars:** `ADOBE_AJO_TOKEN`, `ADOBE_IMS_ORG`

---

## Files to Create / Modify

| File | Action |
|------|--------|
| `backend/integrations/supabase_client.py` | CREATE |
| `backend/integrations/macro_signals.py` | CREATE |
| `backend/integrations/adobe_ajo.py` | CREATE |
| `backend/core/serializer.py` | MODIFY — Supabase dual-write |
| `backend/regime/detector.py` | MODIFY — macro override |
| `backend/execution/loop.py` | MODIFY — AJO pause/scale |
| `backend/decision/scoring.py` | MODIFY — competition penalty |
| `backend/api.py` | MODIFY — intelligence loop tick + cac_estimate |
| `.env.example` | MODIFY — add all new env vars |
| `requirements.txt` | MODIFY — add `supabase` |

---

## End-to-End Loop After These Changes

```
[Startup] Load state: Supabase → DuckDB fallback
     ↓
[Every 6s] run_cycle():
  decide()  → world_model + causal_score (+ competition_penalty) + bandit_weight
              + regime_bonus + velocity + acceleration → apply_confidence
     ↓
  execute() → CVXPY budget → simulate ROAS
              → auto_kill → AJO.pause_campaign() [real platform]
              → amplifier → AJO.scale_campaign() [real platform]
              → strategy_memory, reality_gap
     ↓
  learn()   → calibration, temporal credit, counterfactual eval
  causal    → Granger graph update
  regime    → detect (+ FRED macro override) → transition detection
     ↓
[Every 5 min] scheduler → trend keywords → intelligence_loop → core/memory
[On save]     DuckDB + Supabase dual-write
[API]         /metrics includes cac_estimate
```

---

## Verification Checklist

- [ ] `pip install supabase && python -m pytest tests/ -q` — all tests pass
- [ ] Set `SUPABASE_URL` → restart → `load_state()` returns prior state
- [ ] Set `FRED_API_KEY` → high-VIX period shows `detected_regime: volatile` in logs
- [ ] Set `ADOBE_AJO_TOKEN` → killed campaigns appear as "stopped" in AJO
- [ ] `GET /metrics` → includes `cac_estimate` field after 10+ discovery cycles
- [ ] `GET /causal` → shows negative-weight edges for high-competition variants
